extensions/wikidata/src/org/openrefine/wikidata/editing/ConnectionManager.java

```diff
@@ -23,114 +23,100 @@
  ******************************************************************************/
 package org.openrefine.wikidata.editing;
 
-import java.io.IOException;
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.wikidata.wdtk.wikibaseapi.ApiConnection;
 import org.wikidata.wdtk.wikibaseapi.BasicApiConnection;
 import org.wikidata.wdtk.wikibaseapi.LoginFailedException;
+import org.wikidata.wdtk.wikibaseapi.OAuthApiConnection;
 import org.wikidata.wdtk.wikibaseapi.apierrors.MediaWikiApiErrorException;
 
-import com.fasterxml.jackson.databind.node.ArrayNode;
-import com.fasterxml.jackson.databind.node.ObjectNode;
-import com.google.refine.ProjectManager;
-import com.google.refine.preference.PreferenceStore;
-import com.google.refine.util.ParsingUtilities;
+import java.io.IOException;
 
 /**
- * Manages a connection to Wikidata, with login credentials stored in the
- * preferences.
- * 
- * Ideally, we should store only the cookies and not the password. But
- * Wikidata-Toolkit does not allow for that as cookies are kept private.
- * 
- * This class is also hard-coded for Wikidata: generalization to other Wikibase
- * instances should be feasible though.
- * 
+ * Manages a connection to Wikidata.
+ * <p>
+ * The connection can be either {@link BasicApiConnection} or {@link OAuthApiConnection}.
+ * <p>
+ * This class is also hard-coded for Wikidata,
+ * it will be generalized to other Wikibase instances soon.
+ *
  * @author Antonin Delpeuch
+ * @author Lu Liu
  */
 
 public class ConnectionManager {
-    
-    final static Logger logger = LoggerFactory.getLogger("connection_mananger");
 
-    public static final String PREFERENCE_STORE_KEY = "wikidata_credentials";
```

Comment:
For people upgrading from previous versions, should we try to delete any existing preference setting stored at this key? That would ensure we clean up credentials stored in plain text there.